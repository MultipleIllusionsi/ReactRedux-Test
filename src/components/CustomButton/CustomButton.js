import React from "react";
import { Link } from "react-router-dom";
import PropTypes from "prop-types";

import "./CustomButton.scss";

const CustomButton = ({ color, text, to, className, onClick }) => (
  <>
    {to ? (
      <button className={`custom-btn ${className} ${color}`}>
        <Link to={to}>{text}</Link>
      </button>
    ) : (
      <button onClick={onClick} className={`custom-btn ${color}`}>
        {text}
      </button>
    )}
  </>
);

CustomButton.propTypes = {
  color: PropTypes.string,
  text: PropTypes.string,
  to: PropTypes.string,
  className: PropTypes.string,
  onClick: PropTypes.func
};

export default CustomButton;
