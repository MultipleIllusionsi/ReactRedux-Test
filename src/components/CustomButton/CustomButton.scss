.green {
    background-color: #009432;

    &:hover {
        background-color: #017929;
    }
}

.red {
    background-color: #e84118;

    &:hover {
        background-color: #c23616;
    }
}

.blue {
    background-color: #686de0;

    &:hover {
        background-color: #4834d4;
    }
}

.black {
    background-color: rgb(70, 70, 70);
    color: white;
}

.buttons {
    width: 20%;
    display: flex;
    flex-direction: column;
    justify-content: flex-start;

    & button {
        font-size: 1.7rem;
        color: white;
        padding: 1rem;
        border: none;
        transition: 0.3s all ease-out;
        cursor: pointer;
    }
}