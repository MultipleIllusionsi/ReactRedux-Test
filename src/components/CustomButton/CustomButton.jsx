import React from "react";
import { Link } from "react-router-dom";

import "./CustomButton.scss";

const CustomButton = ({ color, text, to, onClick }) => (
  <>
    {to ? (
      <button className={color}>
        <Link to={to}>{text}</Link>
      </button>
    ) : (
      <button onClick={onClick} className={color}>
        {text}
      </button>
    )}
  </>
);

export default CustomButton;
