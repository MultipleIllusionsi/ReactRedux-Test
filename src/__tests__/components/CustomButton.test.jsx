import { shallow } from "enzyme";
import React from "react";
import CustomButton from "../../components/CustomButton/CustomButton";

it("expect to render CustomButton-link", () => {
  const component = shallow(
    <CustomButton
      className="go-back"
      to="/posts"
      color="black"
      text="Go back"
    />
  );
  expect(component.debug()).toMatchSnapshot();
});

it("expect to render CustomButton-handler", () => {
  const component = shallow(
    <CustomButton onClick={() => {}} color="red" text="Delete" />
  );
  expect(component.debug()).toMatchSnapshot();
});
