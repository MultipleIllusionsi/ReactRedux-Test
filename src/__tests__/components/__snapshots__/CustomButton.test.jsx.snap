// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render CustomButton-handler 1`] = `
"<Fragment>
  <button onClick={[Function: onClick]} className=\\"red\\">
    Delete
  </button>
</Fragment>"
`;

exports[`expect to render CustomButton-link 1`] = `
"<Fragment>
  <button className=\\"go-back black\\">
    <Link to=\\"/posts\\">
      Go back
    </Link>
  </button>
</Fragment>"
`;
